tools/deviceGen/DeviceGen.py

```diff
@@ -0,0 +1,273 @@
+
+                    line = line_format.format(*[formatValue(dict[param]) for param in line_format_params])
+                    res.append(line)
+                except ContinueEx:
+                    continue
+
+def buildJITTraitsHeaderHelper(res, name, json_data, json_type, *line_format_params):
+    sectionHeaders = {
+        "vulkan10core": "VK 1.0",
+        "vulkan11core": "VK 1.1",
+        "vulkan12core": "VK 1.2",
+        "vulkan13core": "VK 1.3",
+        "nablacore": "Nabla Core Extensions",
+        "vulkanext": "Extensions",
+        "nabla": "Nabla"
+    }
+
+    res.append(f"// {name}")
+    for sectionName, sectionContent in json_data.items():
+        if sectionName == "constexprs":
+            continue
+        if sectionName in sectionHeaders:
+            res.append(f"// {sectionHeaders[sectionName]}")
+        for dict in sectionContent:
+            line_format="NBL_CONSTEXPR_STATIC_INLINE {} {} = )===\" + CJITIncludeLoader::to_string({}.{}) + R\"===(;"
+            if 'type' in dict:
+                try:
+                    dict["json_type"] = json_type 
+                    expose = "expose" in dict and dict["expose"] or "expose" not in dict
+                    if not expose:
+                        continue
+
+                    for param in line_format_params:
+                        if param not in dict:
+                            raise ContinueEx
+                    line = line_format.format(*[formatValue(dict[param]) for param in line_format_params])
+                    res.append(line)
+                except ContinueEx:
+                    continue
+
+
+def buildTraitsHeader(**params):
+    res = []
+
+    buildTraitsHeaderHelper(
+        res,
+        f"Limits {params['type']}",
+        params["limits_json"],
+        params["template"],
+        *params['format_params']
+    )
+    res.append(emptyline)
+    buildTraitsHeaderHelper(
+        res,
+        f"Features {params['type']}",
+        params["features_json"],
+        params["template"],
+        *params['format_params']
+    )
+
+    return res
+
+def buildJITTraitsHeader(**params):
+    res = [
+        "std::string jit_traits = R\"===("
+    ]
+
+    buildJITTraitsHeaderHelper(
+        res,
+        f"Limits {params['type']}",
+        params["limits_json"],
+        "limits",
+        *params['format_params']
+    )
+    res.append(emptyline)
+    buildJITTraitsHeaderHelper(
+        res,
+        f"Features {params['type']}",
+        params["features_json"],
+        "features",
+        *params['format_params']
+    )
+
+    res.append(")===\";")
+
+    return res
+
+def writeHeader(file_path, header_builder, **params):
+    try:
+        with open(file_path, mode="w") as file:
+            device_header = header_builder(**params)
+            for line in device_header:
+                file.write(line + '\n')
+    except Exception as ex:
+        print(f"Error while writing to file: {file_path}\nException: {ex}")
```

Comment:
remember to throw an error after handling the exception, custom command should fail at this case and the python script should return non 0 exit level

Reply:
Done

Reply:
@AnastaZIuk resolve if done